features/opennms-es-rest/main-module/src/main/java/org/opennms/plugins/elasticsearch/rest/RestClientFactory.java

```diff
@@ -122,13 +131,23 @@ public void setServers(Set<String> servers) {
 	 * @param esusername Optional HTTP username
 	 * @param espassword Optional HTTP password
 	 */
-	public RestClientFactory(String elasticSearchURL, String esusername, String espassword ){
+	public RestClientFactory(String elasticSearchURL, String esusername, String espassword ) throws MalformedURLException {
 
 		// If multiple URLs are specified in a comma-separated string, split them up
-		config = new HttpClientConfig.Builder(Arrays.asList(elasticSearchURL.split(",")))
-				.multiThreaded(true)
-				.defaultCredentials(esusername, espassword)
-				.build();
+		final List<String> targetHosts = Arrays.asList(elasticSearchURL.split(","));
+		final HttpClientConfig.Builder configBuilder = new HttpClientConfig.Builder(targetHosts).multiThreaded(true);
+		if (!Strings.isNullOrEmpty(esusername) && !Strings.isNullOrEmpty(espassword)) {
+			final URL targetUrl = new URL(targetHosts.get(0));
+
+			if (targetHosts.size() > 1) {
+				LOG.warn("Username/Password have been defined, but multiple target hosts were found. Each host will use the same credentials.");
+				LOG.warn("Username/Password have been defined, but multiple target hosts were found. Preemptive auth is only enabled for host {}",  targetHosts.get(0));
+			}
```

Comment:
Is there a reason you have 2 almost identical `LOG.warn` calls one right after the other?

Reply:
Yes, because I like it this way (-: